core/pkg/eval/json_evaluator.go

```diff
@@ -252,7 +251,7 @@ func (je *JSONEvaluator) evaluateVariant(
 		err = jsonlogic.Apply(bytes.NewReader(targetingBytes), bytes.NewReader(b), &result)
 		if err != nil {
 			je.Logger.ErrorWithID(reqID, fmt.Sprintf("error applying rules: %s", err))
-			return "", model.ErrorReason, err
+			return "", model.ErrorReason, errors.New(model.FlagDisabledErrorCode)
```

Comment:
This is a json logic applying error

```suggestion
			return "", model.ErrorReason, errors.New(model.ParseErrorCode)
```

Reply:
Agreed with @Kavindu-Dodan on this one.